.github/workflows/cargo_llvm_cov.yml

```diff
@@ -47,139 +49,143 @@ jobs:
         uses: taiki-e/install-action@7e58f89e24a544d88f7a74c6eed8a3df3fd4c658 # pin@nextest
 
       - name: Set Swap Space
-        uses: pierotofy/set-swap-space@49819abfb41bd9b44fb781159c033dba90353a7c # pin@master
+        uses: actionhippie/swap-space@v1
         with:
-          swap-size-gb: 256
-
-      - name: Install Rust
-        run: rustup update stable
+          size: 256G
 
-      - name: Run code coverage for nextest
-        run: IOTA_SKIP_SIMTESTS=1 cargo llvm-cov --ignore-run-fail --lcov --output-path lcov.info nextest
-
-      - name: Upload report to Codecov for nextest
-        uses: codecov/codecov-action@b9fd7d16f6d7d1b5d2bec1a2887e65ceed900238 # pin@v4
-        with:
-          files: lcov.info
+      - name: Install llvm-tools
+        run: |
+          TOOLCHAIN=$(rustup show active-toolchain | cut -d ' ' -f 1)
+          rustup component add llvm-tools --toolchain "$TOOLCHAIN"
+          LLVM_PROFDATA="$HOME/.rustup/toolchains/$TOOLCHAIN/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-profdata"
+
+          if [ -x "$LLVM_PROFDATA" ]; then
+            $LLVM_PROFDATA --version
+            echo "LLVM_PROFDATA=$LLVM_PROFDATA" >> $GITHUB_ENV
+          else
+            echo "🚨 Error 🚨: llvm-profdata not found at $LLVM_PROFDATA, or is not executable."
+            exit 1
+          fi
+
+      - name: Install sccache
+        run: |
+          cargo install sccache
+          echo -e "\n[build]\nrustc-wrapper = \"sccache\"" >> $HOME/.cargo/config.toml
+          echo "sccache: $(which sccache)" 
+          cat "$HOME/.cargo/config.toml"
```

Comment:
I think I would remove this and open an issue. Surely there are GH actions to do this in a more efficient and elegant way. This has now changed settings and it's a bit annoying to revert.

Reply:
done